buildenv/jenkins/jobs/infrastructure/Update-Reference-Repos

```diff
@@ -0,0 +1,143 @@
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+SETUP_LABEL = params.SETUP_LABEL
+if (!SETUP_LABEL) {
+    SETUP_LABEL = 'worker'
+}
+
+LABEL = params.LABEL
+if (!LABEL) {
+    LABEL = 'ci.role.build'
+}
+
+properties([buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10')),
+            pipelineTriggers([cron('''# Daily at 11:00pm
+                                        0 23 * * *''')])])
+
+def jobs = [:]
+
+timeout(time: 6, unit: 'HOURS') {
+    timestamps {
+        node(SETUP_LABEL) {
+            try{
+                checkout scm
+                def variableFile = load 'buildenv/jenkins/common/variables-functions'
+                variableFile.parse_variables_file()
+                variableFile.set_user_credentials()
+
+                def osLabels = ['sw.os.aix', 'sw.os.linux', 'sw.os.osx', 'sw.os.windows']
+
+                for (aNode in jenkins.model.Jenkins.instance.getLabel(LABEL).getNodes()) {
+                    def nodeName = aNode.getDisplayName()
+
+                    if (aNode.toComputer().isOffline()) {
+                        // skip offline slave
+                        continue
+                    }
+
+                    def foundLabel = false
+                    def nodeLabels = aNode.getLabelString().tokenize(' ')
+                    for (osLabel in osLabels) {
+                        if (nodeLabels.contains(osLabel)) {
+                            foundLabel = true
+                            break
+                        }
+                    }
+
+                    // get Eclipse OpenJ9 extensions repositories from variables file
+                    def repos = get_openjdk_repos(VARIABLES.openjdk, foundLabel)
+
+                    jobs["${nodeName}"] = {
+                        node("${nodeName}"){
+                            stage("${nodeName} - Update Reference Repo") {
+                                refresh(nodeName, "${HOME}/openjdk_cache", repos, foundLabel)
+                            }
+                        }
+                    }
+                }
+            } finally {
+                cleanWs()
+            }
+        }
+
+        parallel jobs
+    }
+}
+
+/*
+* Creates and updates the git reference repository cache on the node.
+*/
+def refresh(node, cacheDir, repos, isKnownOs) {
+    sh "mkdir -p ${cacheDir}"
```

Comment:
Can we add a bool param here to rm cacheDir if requested at launch time?

Reply:
Added CLEAN_CACHE_DIR parameter.